trigger-actions-framework/main/default/classes/ActionUtility.cls

```diff
@@ -0,0 +1,75 @@
+/**
+ * Utilities for managing actions.
+ *
+ * Allows for the retrieval of the current user's permissions only once during the entire transaction
+ * Consolidates code that is shared between MetadataTriggerHandler and FinalizerHandler classes
+ */
+
+public with sharing class ActionUtility {
+    @TestVisible
+    public static Set<String> userPermissions {
+        get {
+            if (userPermissions == null) {
+                userPermissions = new Set<String>();
+
+                List<PermissionSetAssignment> permSets = [
+                    SELECT AssigneeId, Assignee.FirstName, PermissionSetId, PermissionSet.Name
+                    FROM PermissionSetAssignment
+                    WHERE AssigneeId = :UserInfo.getUserId()
+                ];
+                Set<Id> permSetIds = new Set<Id>();
+                for (PermissionSetAssignment psa : permSets) {
+                    permSetIds.add(psa.PermissionSetId);
+                }
+                List<CustomPermission> customPerms = [
+                    SELECT DeveloperName, MasterLabel, NamespacePrefix
+                    FROM CustomPermission
+                    WHERE Id IN (SELECT SetupEntityId FROM SetupEntityAccess WHERE ParentId IN :permSetIds)
+                ];
+                for (CustomPermission cp : customPerms) {
+                    String qualifiedPerm = String.isBlank(cp.NamespacePrefix)
+                        ? cp.DeveloperName
+                        : cp.NamespacePrefix + '__' + cp.DeveloperName;
+                    userPermissions.add(qualifiedPerm);
+                }
+            }
+            return userPermissions;
+        }
+        private set;
+    }
+
+    public static boolean isNotBypassed(String requiredPermission, String bypassPermission) {
```

Comment:
Curious why using isNOTBypassed - it can be confusing to the user/dev if Booleans are expressed negatively.

Gentle suggestion: Update to `isBypassed` and return the negative of the output (which is already a negative, I see).